source/Calamari.Tests/KubernetesFixtures/ResourceStatus/Resources/PodTests.cs

```diff
@@ -154,10 +155,10 @@ public void WhenContainerFailsExecutionAndRestarting_TheStatusShouldShowCrashLoo
                 })
                 .Build();
     
-            var pod = (Pod)ResourceFactory.FromJson(podResponse);
+            var pod = (Pod)ResourceFactory.FromJson(podResponse, new Options());
 
             pod.Status.Should().Be("CrashLoopBackOff");
-            pod.ResourceStatus.Should().Be(KubernetesResources.ResourceStatus.Failed);
+            pod.ResourceStatus.Should().Be(KubernetesResources.ResourceStatus.Successful);
```

Comment:
Because `CrashLoopBackOff` state happens after the Pod has reached the `Running` phase, in the simplified logic we'll report the status as successful. This is not ideal, but I'm refraining from making the logic too complicated. The `CrashLoopBackOff` state will still be displayed, and the controlling object such as ReplicaSet will not have this Pod as one of the ready Pods, so the overall deployment will not succeed anyway. The only exception is when the user is trying to deploy a Pod directly from YAML, which is a discouraged use case. In this case the deployment will be successful while the Pod is in a `CrashLoopBackOff`.

Reply:
To to be clear, if we a pod goes into `CrashLoopBackOff` then the K8S cluster considers it to be `Running`? Thats surprising. I would have thought that it would still be failed. So from the users perspective on the status page we will report this pod as successful (even if the replica set isnt yet successful)

Reply:
Yes it is. From this diagram, you can see the `CrashLoopBackOff` state happens when the Pod enters a `Running` phase.

Reply:
Forgot to put in the diagram link: https://miro.com/app/board/uXjVPkFe6vA=/